e2e/ci-e2e/tests/ci.e2e.test.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  copyFile,
+  mkdir,
+  readFile,
+  rename,
+  rm,
+  writeFile,
+} from 'node:fs/promises';
+import { dirname, join } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import {
+  type DiffResult,
+  type FetchResult,
+  type SimpleGit,
+  simpleGit,
+} from 'simple-git';
+import {
+  type Comment,
+  type GitRefs,
+  type Options,
+  type ProviderAPIClient,
+  type RunResult,
+  runInCI,
+} from '@code-pushup/ci';
+
+describe('CI package', () => {
+  const fixturesDir = join(
+    fileURLToPath(dirname(import.meta.url)),
+    '..',
+    'mocks',
+    'fixtures',
+  );
+  const workDir = join(process.cwd(), 'tmp', 'ci-e2e', 'test-repo');
+  const outputDir = join(workDir, '.code-pushup');
+
+  const options = {
+    directory: workDir,
+  } satisfies Options;
+
+  let git: SimpleGit;
+
+  beforeEach(async () => {
+    await rm(workDir, { recursive: true, force: true });
+    await mkdir(workDir, { recursive: true });
+    await copyFile(
+      join(fixturesDir, 'code-pushup.config.ts'),
+      join(workDir, 'code-pushup.config.ts'),
+    );
+    await writeFile(join(workDir, 'index.js'), 'console.log("Hello, world!")');
+
+    git = simpleGit(workDir);
+
+    vi.spyOn(git, 'fetch').mockResolvedValue({} as FetchResult);
+    vi.spyOn(git, 'diffSummary').mockResolvedValue({
+      files: [{ file: 'index.ts', binary: false }],
+    } as DiffResult);
+    vi.spyOn(git, 'diff').mockResolvedValue('');
+
+    await git.init();
+    await git.addConfig('user.name', 'John Doe');
+    await git.addConfig('user.email', 'john.doe@example.com');
+    await git.branch(['-M', 'main']);
```

Comment:
We have helper functions to setup git repos. 
Have a look at: [testing/test-utils/src/lib/utils/git.ts](https://github.com/code-pushup/cli/blob/main/testing/test-utils/src/lib/utils/git.ts)

Reply:
Used the shared utils and refactored them a little.